File: Lab1/lab1_sequential.py
import numpy as np

def main():
    Lx: float = 10.0
    Ly: float = 10.0
    Nx: int = 20
    Ny: int = 20
    g: float = 1.0
    lambda_: float = 1.0

    dx = Lx / (Nx - 1)
    dy = Ly / (Ny - 1)

    T = np.zeros((Ny, Nx))

    # Warunki brzegowe
    T[:, 0] = 0
    T[:, -1] = 0
    T[0, :] = 0
    T[-1, :] = 0

    tolerance: float = 1e-6
    error: float = 1.0

    while error > tolerance:
        T_old = T.copy()

        for j in range(1, Ny - 1):
            for i in range(1, Nx - 1):
                T[j, i] = (T_old[j + 1, i] + T_old[j - 1, i] + 
                        T_old[j, i + 1] + T_old[j, i - 1] - 
                        (g / lambda_) * (dx ** 2 + dy ** 2)) / 4.0
                
        error: float = np.max(np.abs(T - T_old))

    print(T)

if __name__ == "__main__":
    main()

File: Lab1/lab1_c/lab1_sequential.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>

#define Nx 20
#define Ny 20

int main() {
    double Lx = 10.0;
    double Ly = 10.0;
    double g = 1.0;
    double lambda_ = 1.0;
    double dx = Lx / (Nx - 1);
    double dy = Ly / (Ny - 1);
    double tolerance = 1e-6;
    double error = 1.0;

    double T[Ny][Nx] = {0};

    // Warunki brzegowe
    for (int i = 0; i < Nx; i++) {
        T[0][i] = 0.0;
        T[Ny-1][i] = 0.0;
    }
    for (int j = 0; j < Ny; j++) {
        T[j][0] = 0.0;
        T[j][Nx-1] = 0.0;
    }

    double T_old[Ny][Nx] = {0};

    while (error > tolerance) {
        for (int j = 0; j < Ny; j++) {
            for (int i = 0; i < Nx; i++) {
                T_old[j][i] = T[j][i];
            }
        }

        for (int j = 1; j < Ny - 1; j++) {
            for (int i = 1; i < Nx - 1; i++) {
                T[j][i] = (T_old[j + 1][i] + T_old[j - 1][i] + 
                           T_old[j][i + 1] + T_old[j][i - 1] - 
                           (g / lambda_) * (dx * dx + dy * dy)) / 4.0;
            }
        }

        error = 0.0;
        for (int j = 1; j < Ny - 1; j++) {
            for (int i = 1; i < Nx - 1; i++) {
                double diff = fabs(T[j][i] - T_old[j][i]);
                if (diff > error) {
                    error = diff;
                }
            }
        }
    }

    for (int j = 0; j < Ny; j++) {
        for (int i = 0; i < Nx; i++) {
            printf("%6.2f ", T[j][i]);
        }
        printf("\n");
    }

    return 0;
}